sstables/mx/reader.cc

```diff
@@ -2104,8 +2104,8 @@ future<uint64_t> validate(
     validating_consumer consumer(schema, permit, sstable, std::move(error_handler));
     auto context = data_consume_rows<data_consume_rows_context_m<validating_consumer>>(*schema, sstable, consumer, integrity_check::yes);
 
-    std::optional<sstables::index_reader> idx_reader;
-    idx_reader.emplace(sstable, permit, tracing::trace_state_ptr{}, sstables::use_caching::no, false);
+    auto idx_reader = std::make_unique<index_reader>(sstable, permit, tracing::trace_state_ptr{}, sstables::use_caching::no, false);
+    auto big_index_reader = dynamic_cast<index_reader*>(idx_reader.get());
```

Comment:
How are idx_reader and big_index_reader different? Perhaps you wanted to downgrade idx_reader to abstract_index_reader?

You could reverse the flow, create index_reader, and implicit-convert it to abstract_index_reader.

I think the names confused you, index_reader is also an abstract name. We'd be better off having index_reader and {big/trie}_index_reader to make everything clear. But such noisy renames are best done after the dust settles.